packages/@aws-cdk/aws-stepfunctions-tasks/lib/athena/start-query-execution.ts

```diff
@@ -91,24 +91,30 @@ export class AthenaStartQueryExecution extends sfn.TaskStateBase {
       }),
     ];
 
+    policyStatements.push(
+      new iam.PolicyStatement({
+        actions: ['s3:CreateBucket'],
+        resources: ['*'], // Need * permissions to create new output location https://docs.aws.amazon.com/athena/latest/ug/security-iam-athena.html
+      }),
+    );
+
     policyStatements.push(
       new iam.PolicyStatement({
         actions: ['s3:AbortMultipartUpload',
-          's3:CreateBucket',
           's3:GetBucketLocation',
           's3:GetObject',
           's3:ListBucket',
           's3:ListBucketMultipartUploads',
           's3:ListMultipartUploadParts',
           's3:PutObject'],
-        resources: [this.props.resultConfiguration?.outputLocation ?? '*'], // Need S3 location where data is stored https://docs.aws.amazon.com/athena/latest/ug/security-iam-athena.html
+        resources: [this.props.resultConfiguration?.outputLocation ? 'arn:aws:s3:::' + this.props.resultConfiguration?.outputLocation.slice(5) + '*' : '*'], // Need S3 location where data is stored or Athena throws an Unable to verify/create output bucket https://docs.aws.amazon.com/athena/latest/ug/security-iam-athena.html
```

Comment:
should the type of s3 location be  `s3.Location` instead `https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-s3.Location.html`- it would avoid having to do this `slice` and is more consistent with how s3 locations are used elsewhere within the CDK

Reply:
Yes, will import and switch.